src/overlays/actors/ovl_En_Diving_Game/z_en_diving_game.c

```diff
@@ -39,57 +67,522 @@ static ColliderCylinderInit D_809EF0B4 = {
+            func_80106CCC(globalCtx);
+            if (!(gSaveContext.eventChkInf[3] & 0x100)) {
+                func_80088B34(BREG(2) + 50);
+            } else {
+                func_80088B34(BREG(2) + 50);
+            }
+            func_800F5ACC(0x6C);
+            func_8002DF54(globalCtx, NULL, 7);
+            this->actor.textId = 0x405B;
+            this->unk_292 = 5;
+            this->unk_2A8 = 2;
+            this->actionFunc = func_809EDD4C;
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Diving_Game/func_809EDD4C.s")
+void func_809EE8F0(EnDivingGame* this, GlobalContext* globalCtx) {
+    SkelAnime_Update(&this->skelAnime);
+    if ((this->unk_292 == func_8010BDBC(&globalCtx->msgCtx) && (func_80106BC8(globalCtx) != 0))) {
+        func_80106CCC(globalCtx);
+        this->actionFunc = func_809EDD4C;
+    } else {
+        func_809EDB08(this, globalCtx);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Diving_Game/func_809EDEDC.s")
+void func_809EE96C(EnDivingGame* this, GlobalContext* globalCtx) {
+    SkelAnime_Update(&this->skelAnime);
+    if ((this->unk_292 == func_8010BDBC(&globalCtx->msgCtx) && (func_80106BC8(globalCtx) != 0))) {
+        func_80106CCC(globalCtx);
+        func_8002DF54(globalCtx, NULL, 7);
+        this->actor.textId = 0x4056;
+        this->unk_292 = 5;
+        this->unk_2A8 = 1;
+        this->actionFunc = func_809EDD4C;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Diving_Game/func_809EE048.s")
+void func_809EEA00(EnDivingGame* this, GlobalContext* globalCtx) {
+    SkelAnime_Update(&this->skelAnime);
+    if ((this->unk_292 == func_8010BDBC(&globalCtx->msgCtx) && (func_80106BC8(globalCtx) != 0))) {
+        func_80106CCC(globalCtx);
+        this->actor.parent = NULL;
+        func_8002F434(&this->actor, globalCtx, GI_SCALE_SILVER, 90.0f, 10.0f);
+        this->actionFunc = func_809EEA90;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Diving_Game/func_809EE0FC.s")
+void func_809EEA90(EnDivingGame* this, GlobalContext* globalCtx) {
+    SkelAnime_Update(&this->skelAnime);
+    if (Actor_HasParent(&this->actor, globalCtx)) {
+        this->actionFunc = func_809EEAF8;
+    } else {
+        func_8002F434(&this->actor, globalCtx, GI_SCALE_SILVER, 90.0f, 10.0f);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Diving_Game/func_809EE194.s")
+void func_809EEAF8(EnDivingGame* this, GlobalContext* globalCtx) {
+    SkelAnime_Update(&this->skelAnime);
+    if (func_8010BDBC(&globalCtx->msgCtx) == 6) {
+        if (func_80106BC8(globalCtx) != 0) {
+            // "Successful completion"
+            osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ 正常終了 ☆☆☆☆☆ \n" VT_RST);
+            this->unk_31E = this->unk_2A8 = this->unk_29C = this->unk_2A2 = this->unk_2A4 = 0;
+            gSaveContext.eventChkInf[3] |= 0x100;
+            this->actionFunc = func_809EDCB0;
+        }
+    }
```

Comment:
```suggestion
    if ((func_8010BDBC(&globalCtx->msgCtx) == 6) && (func_80106BC8(globalCtx) != 0)) {
            // "Successful completion"
            osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ 正常終了 ☆☆☆☆☆ \n" VT_RST);
            this->unk_31E = this->unk_2A8 = this->unk_29C = this->unk_2A2 = this->unk_2A4 = 0;
            gSaveContext.eventChkInf[3] |= 0x100;
            this->actionFunc = func_809EDCB0;
    }
```

with the indenting fixed